changelog/2645.bugfix.rst

```diff
@@ -0,0 +1,3 @@
+Fixed an issue where an ``HTTPConnection`` instance would erroneously reuse the socket
```

Comment:
We should delete this file since there's no `changelog/` directory on the 1.26.x branch.

Reply:
Done